README.md

```diff
@@ -16,7 +16,384 @@ SPDX-License-Identifier: MIT
+    func configure() {
+        do {
+            try scheduler.createOrUpdateTask(
+                id: "my-daily-task",
+                title: "Daily Questionnaire",
+                instructions: "Please fill out the Questionnaire every day.",
+                category: .questionnaire,
+                schedule: .daily(hour: 9, minute: 0, startingAt: .today)
+            )
+        } catch {
+            // handle error (e.g., visualize in your UI)
+        }
+    }
+}
+```
+
+### Creating and Managing Tasks
+
+Then, configure the [`Scheduler`](https://swiftpackageindex.com/stanfordspezi/spezischeduler/documentation/spezischeduler/scheduler) module and your custom module in your `SpeziAppDelegate`:
+```swift
+class ExampleAppDelegate: SpeziAppDelegate {
+    override var configuration: Configuration {
+        Configuration(standard: ExampleStandard()) {
+            MySchedulerModule()
+            Scheduler()
+        }
+    }
+}
+```
+
+### Task Scheduling Options
+
+The Scheduler supports various scheduling patterns using the [`Schedule`](https://swiftpackageindex.com/stanfordspezi/spezischeduler/documentation/spezischeduler/schedule) type:
+
+```swift
+// One-time task
+let onceSchedule = Schedule.once(at: Date(), duration: .tillEndOfDay)
+
+// Daily tasks
+let dailySchedule = Schedule.daily(hour: 8, minute: 30, startingAt: .today)
+
+// Weekly tasks
+let weeklySchedule = Schedule.weekly(
+    weekday: .monday, 
+    hour: 10, 
+    minute: 0, 
+    startingAt: .today
+)
+
+// Monthly tasks
+let monthlySchedule = Schedule.monthly(
+    day: 1, 
+    hour: 9, 
+    minute: 0, 
+    startingAt: .today
+)
+
+// Custom recurrence patterns
+var customRule = Calendar.RecurrenceRule.weekly(calendar: .current, end: .never)
+customRule.weekdays = [.every(.monday), .every(.wednesday), .every(.friday)]
+let customSchedule = Schedule(startingAt: .today, recurrence: customRule)
+```
+
+### Task Categories and Metadata
+
+Tasks can include categories and additional metadata for better organization and functionality:
+
+```swift
+try scheduler.createOrUpdateTask(
+    id: "medication-reminder",
+    title: "Morning Medication",
+    instructions: "Take your prescribed morning medication with water.",
+    category: .medication,
+    schedule: .daily(hour: 8, minute: 0, startingAt: .today),
+    tags: ["health", "medication", "daily"]
+) { context in
+    // Store additional metadata using the @Property macro
+    context.about = "Take your daily medication as prescribed by your healthcare provider."
+}
+```
+
+### Notifications
+
+For basic notification support, you can use the [`Notifications`](https://github.com/StanfordSpezi/SpeziNotifications) module in your app configuration:
+
+```swift
+class ExampleAppDelegate: SpeziAppDelegate {
+    override var configuration: Configuration {
+        Configuration(standard: ExampleStandard()) {
+            MySchedulerModule()
+            Scheduler()
+            Notifications()
+        }
+    }
+}
+```
+
```

Comment:
IMO this is a little misleading; the app doesn't need to include the `Notifications()` module in its configuration to get notification support

maybe instead we should just have an example of a `createOrUpdateTask` call that specifies the notification-related parameters, and have some text explaining what they do?

Reply:
I noticed that the Template App includes `Notifications()` in the configuration which is why I included it here. Is there some other reason it's being included there?

See: https://github.com/StanfordSpezi/SpeziTemplateApplication/blob/28fa049c4bfdccef125c9e3e387ef35d299049e1/TemplateApplication/TemplateApplicationDelegate.swift#L54